src/reset.cpp

```diff
@@ -197,4 +197,25 @@ ResetPtr Reset::clone() const
     return r;
 }
 
+bool Reset::doEqual(const EntityPtr &other) const
+{
+    if (Entity::doEqual(other)) {
+        auto reset = std::dynamic_pointer_cast<Reset>(other);
+        if (reset != nullptr && mPimpl->mOrder == reset->order() && mPimpl->mResetValue == reset->resetValue() && mPimpl->mResetValueId == reset->resetValueId() && mPimpl->mTestValue == reset->testValue() && mPimpl->mTestValueId == reset->testValueId()) {
+            bool equal = true;
+            if (mPimpl->mTestVariable != nullptr && !mPimpl->mTestVariable->equal(reset->testVariable())) {
```

Comment:
I think the performance gains here would be so small that it would not be useful.  If you are looking to write performant efficient programs don't write them in C++.